### PR TITLE
* Adds removeAll:  removeAllFoundIn: and removeAllSuchThat: to the colle...

### PR DESCRIPTION
**Pull request by sebastianconcept**
...ction API

* Adds one test for each of those methods.
* Upgrades dependency on amber-dev to 0.3.1

To satisfy:
https://github.com/amber-smalltalk/amber/issues/1144
Collection removeAll: ; removeAllFoundIn: ; removeAllSuchThat:

### CODE REVIEW

File: CHANGELOG [hunk 1]
```diff
@@ -1,3 +1,13 @@
+21 January 2015 - Release 0.14.4
```
Comment (herby) [inline CHANGELOG]: You want to have release for this?

File: CHANGELOG [hunk 2]
```diff
@@ -1,3 +1,13 @@
...
+===================================
+
+Highlights:
+
+* Adds removeAll:  removeAllFoundIn: and removeAllSuchThat: to the collection API
```
Comment (herby) [inline CHANGELOG]: This should not be here.

File: CHANGELOG [hunk 3]
```diff
@@ -1,3 +1,13 @@
...
+* Adds one test for each of those methods.
```
Comment (herby) [inline CHANGELOG]: This neither. CHANGELOG only contains general description of what the release contains.
API is is in API-CHANGES.txt.

File: package.json [hunk 4]
```diff
@@ -29,7 +29,7 @@
     "test": "grunt test && cd external && cd amber-cli && npm test && cd .. && cd .."
   },
   "devDependencies": {
-    "amber-dev": "^0.3.0",
+    "amber-dev": "^0.3.1",
```
Comment (herby) [inline package.json]: Not needed, really. Worked fine with 0.3.0 and anyone getting the clone or doing `bower install` gets 0.3.1 (or newer 0.3.x) anyway. Don't include.

File: src/Kernel-Tests.st [hunk 5]
```diff
@@ -1167,6 +1178,44 @@ testLastN
...
+	
+	original := #( 1 2 3 4 5 6 7 8 9 ).
+	part := #( 2 4 6 ).
+	
+	original removeAll: part.
+
+	self shouldnt: [ self assert: (part allSatisfy: [ :e | (original includes: e) not ]) ] raise: Error.
+	self should: [ original removeAll: #( 11 12 ) ] raise: Error
+!
+
```
Comment (herby) [inline src/Kernel-Tests.st]: Those are suboptimal tests. There are many types of collections. There are tests that are written more generically and are inherited, thus testing many types of collections. Look at how other tests are done for Collection hierarchy.

File: CHANGELOG (hunk 1 above)
Comment (sebastianconcept) [inline CHANGELOG]: Nothing special.

I actually don't know how the process work. I've just put there that to have a place to write the highlights.

You don't do it that what? How do you do it?

Comment (herby) [inline CHANGELOG]: I was freaked out by the date, it looked as if you plan to release at 21 January. I often write `???` there if the date is not known.

Comment (sebastianconcept) [inline CHANGELOG]: haha oh nooo. Sorry for that.

Glad to know, will be `???` for the next time.

**Comment by herby** [comment]
Stale. Closing.

**Pull request closed without merging**